container/src/main/java/com/flipkart/poseidon/internal/ByteArrayDataType.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2015 Flipkart Internet, pvt ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.poseidon.internal;
+
+import com.flipkart.poseidon.model.annotations.Description;
+import com.flipkart.poseidon.model.annotations.Name;
+import com.flipkart.poseidon.model.annotations.Version;
+
+import flipkart.lego.api.entities.DataType;
+
+@Name("ByteArrayDataType")
+@Version(major = 1, minor = 0, patch = 0)
+@Description("DataType wrapper for byte array")
+public class ByteArrayDataType implements DataType {
+
+	private static final long serialVersionUID = 1L;
+	
+	private byte[] rawBytes;
+	
+    public ByteArrayDataType() {
+        super();
+    }
+
+    public ByteArrayDataType(byte[] rawBytes) {
+    	this.rawBytes = rawBytes;
+    }
+
+	public byte[] getRawBytes() {
+		return rawBytes;
+	}
+
+}
```

Comment:
@regunathb , yes just realized that, though these are internal data types for poseidon use, applications built on poseidon too use them for sending Map and List out of DataSources. A few options to keep backward compatibility (as we follow semantic versioning)

1. Keep these two classes in existing path and add only ByteArrayDataType under container and with next major release (6.x) we can move the remaining classes also to container.

2. We can have a custom check in APIBuildable::build() which is dealing with the DataType returned by OrchestratorDataSource. It can check for ByteArrayDataType and then inject only the raw bytes into [poseidonResponse.setResponse()](https://github.com/flipkart-incubator/Poseidon/blob/master/api/src/main/java/com/flipkart/poseidon/api/APIBuildable.java#L79)